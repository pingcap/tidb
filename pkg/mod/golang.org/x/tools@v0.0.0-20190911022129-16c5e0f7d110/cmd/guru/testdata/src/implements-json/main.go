package main

// Tests of 'implements' query, -output=json.
// See go.tools/guru/guru_test.go for explanation.
// See implements.golden for expected query results.

func main() {
}

type E interface{} // @implements E "E"

type F interface { // @implements F "F"
	f()
}

type FG interface { // @implements FG "FG"
	f()
	g() []int // @implements slice "..int"
}

type C int // @implements C "C"
type D struct{}

func (c *C) f() {} // @implements starC ".C"
func (d D) f()  {} // @implements D "D"

func (d *D) g() []int { return nil } // @implements starD ".D"
