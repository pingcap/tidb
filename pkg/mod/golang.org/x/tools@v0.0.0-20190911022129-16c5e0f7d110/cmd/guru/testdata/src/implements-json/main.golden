-------- @implements E --------
{
	"type": {
		"name": "implements-json.E",
		"pos": "testdata/src/implements-json/main.go:10:6",
		"kind": "interface"
	}
}
-------- @implements F --------
{
	"type": {
		"name": "implements-json.F",
		"pos": "testdata/src/implements-json/main.go:12:6",
		"kind": "interface"
	},
	"to": [
		{
			"name": "*implements-json.C",
			"pos": "testdata/src/implements-json/main.go:21:6",
			"kind": "pointer"
		},
		{
			"name": "implements-json.D",
			"pos": "testdata/src/implements-json/main.go:22:6",
			"kind": "struct"
		},
		{
			"name": "implements-json.FG",
			"pos": "testdata/src/implements-json/main.go:16:6",
			"kind": "interface"
		}
	]
}
-------- @implements FG --------
{
	"type": {
		"name": "implements-json.FG",
		"pos": "testdata/src/implements-json/main.go:16:6",
		"kind": "interface"
	},
	"to": [
		{
			"name": "*implements-json.D",
			"pos": "testdata/src/implements-json/main.go:22:6",
			"kind": "pointer"
		}
	],
	"from": [
		{
			"name": "implements-json.F",
			"pos": "testdata/src/implements-json/main.go:12:6",
			"kind": "interface"
		}
	]
}
-------- @implements slice --------
{
	"type": {
		"name": "[]int",
		"pos": "-",
		"kind": "slice"
	}
}
-------- @implements C --------
{
	"type": {
		"name": "implements-json.C",
		"pos": "testdata/src/implements-json/main.go:21:6",
		"kind": "basic"
	},
	"fromptr": [
		{
			"name": "implements-json.F",
			"pos": "testdata/src/implements-json/main.go:12:6",
			"kind": "interface"
		}
	]
}
-------- @implements starC --------
{
	"type": {
		"name": "*implements-json.C",
		"pos": "testdata/src/implements-json/main.go:21:6",
		"kind": "pointer"
	},
	"from": [
		{
			"name": "implements-json.F",
			"pos": "testdata/src/implements-json/main.go:12:6",
			"kind": "interface"
		}
	]
}
-------- @implements D --------
{
	"type": {
		"name": "implements-json.D",
		"pos": "testdata/src/implements-json/main.go:22:6",
		"kind": "struct"
	},
	"from": [
		{
			"name": "implements-json.F",
			"pos": "testdata/src/implements-json/main.go:12:6",
			"kind": "interface"
		}
	],
	"fromptr": [
		{
			"name": "implements-json.FG",
			"pos": "testdata/src/implements-json/main.go:16:6",
			"kind": "interface"
		}
	]
}
-------- @implements starD --------
{
	"type": {
		"name": "*implements-json.D",
		"pos": "testdata/src/implements-json/main.go:22:6",
		"kind": "pointer"
	},
	"from": [
		{
			"name": "implements-json.F",
			"pos": "testdata/src/implements-json/main.go:12:6",
			"kind": "interface"
		},
		{
			"name": "implements-json.FG",
			"pos": "testdata/src/implements-json/main.go:16:6",
			"kind": "interface"
		}
	]
}
